project/ScalaOptionParser.scala

```diff
@@ -108,7 +108,7 @@ object ScalaOptionParser {
     "-g" -> List("line", "none", "notailcails", "source", "vars"),
     "-target" -> List("jvm-1.5", "jvm-1.6", "jvm-1.7", "jvm-1.8"))
   private def multiChoiceSettingNames = Map[String, List[String]](
-    "-Xlint" -> List("adapted-args", "nullary-unit", "inaccessible", "nullary-override", "infer-any", "missing-interpolator", "doc-detached", "private-shadow", "type-parameter-shadow", "poly-implicit-overload", "option-implicit", "delayedinit-select", "by-name-right-associative", "package-object-classes", "unsound-match", "stars-align"),
+    "-Xlint" -> List("adapted-args", "nullary-unit", "inaccessible", "nullary-override", "infer-any", "missing-interpolator", "doc-detached", "private-shadow", "type-parameter-shadow", "poly-implicit-overload", "option-implicit", "delayedinit-select", "by-name-right-associative", "package-object-classes", "unsound-match", "stars-align", "no-implicit-annotation"),
```

Comment:
Perhaps: `implicit-result-type`

Reply:
I thought about it and I agree the rename is the most consistent option.

Reply:
This still looks unfixed?

Reply:
I can attest that naming these options is hard. I suspect no one will ever request `-Xlint:poly-implicit-overload`. You warn for `implicit-lacks-type`, or `-Xlint:infer-X`, since the warning is triggered by inference. `-Xlint:infer-implicit-type`.